ortb/default.go

```diff
@@ -0,0 +1,76 @@
+package ortb
+
+import (
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/util/ptrutil"
+)
+
+const (
+	DefaultPriceGranularityPrecision  = 2
+	DefaultTargetingIncludeWinners    = true
+	DefaultTargetingIncludeBidderKeys = true
+)
+
+func SetDefaults(r *openrtb_ext.RequestWrapper) error {
+	requestExt, err := r.GetRequestExt()
+	if err != nil {
+		return err
+	}
+
+	requestExtPrebid := requestExt.GetPrebid()
+	if requestExtPrebid != nil {
+		hasChanges := setDefaultsTargeting(requestExtPrebid.Targeting)
+
+		if hasChanges {
+			requestExt.SetPrebid(requestExtPrebid)
+		}
+	}
+
+	return nil
+}
+
+func setDefaultsTargeting(targeting *openrtb_ext.ExtRequestTargeting) bool {
+	if targeting == nil {
+		return false
+	}
+
+	hasChanges := false
+
+	if targeting.PriceGranularity == nil {
+		targeting.PriceGranularity = ptrutil.ToPtr(openrtb_ext.NewPriceGranularityDefault())
+		hasChanges = true
+	} else {
+		if targeting.PriceGranularity.Precision == nil {
+			targeting.PriceGranularity.Precision = ptrutil.ToPtr(DefaultPriceGranularityPrecision)
+			hasChanges = true
+		}
+		hasChanges = hasChanges || setDefaultsPriceGranularityRange(targeting.PriceGranularity.Ranges)
+	}
+
+	if targeting.IncludeWinners == nil {
+		targeting.IncludeWinners = ptrutil.ToPtr(DefaultTargetingIncludeWinners)
+		hasChanges = true
+	}
+
+	if targeting.IncludeBidderKeys == nil {
+		targeting.IncludeBidderKeys = ptrutil.ToPtr(DefaultTargetingIncludeBidderKeys)
+		hasChanges = true
+	}
+
+	return hasChanges
```

Comment:
The change tracking is annoying, but necessary to avoid unneeded marshal calls. Couldn't find a better way to do this in Go.